go/worker/registration/worker.go

```diff
@@ -908,16 +932,19 @@ func (w *Worker) registerNode(epoch beacon.EpochTime, hook RegisterNodeHook) err
 		w.logger.Error("not registering: no runtimes provided while runtimes are required",
 			"node_descriptor", nodeDesc,
 		)
-		return fmt.Errorf("registration: no runtimes provided while runtimes are required")
+
+		err = fmt.Errorf("registration: no runtimes provided while runtimes are required")
```

Comment:
This change and the other two with `gerr` are not needed, unless linter complains.

The following example prints `error #2`.
```
func main() {
	test()
}

func test() (err error) {
	defer func() {
		fmt.Println(err)
	}()
	err = fmt.Errorf("error #1")
	return fmt.Errorf("error #2")
}
```